### PR TITLE
Summary statistics for checkers

### PR DESCRIPTION
**Pull request by cservakt**
It can be used to see the number of reports per checker for `parse` command. Before storing it is helpful to verify with the `--summary` flag, which checkers are generating too many reports in a large report directory. The result is a table that has a checker name and number of reports columns. It shows the results in descending order. 

Example command: `CodeChecker parse reports/ --summary` 

The result table would be:

```
---==== Checkers Summary Statistics ====----
-------------------------------------------------------------------------------
Checker name                                                | Number of reports
-------------------------------------------------------------------------------
readability-avoid-const-params-in-decls                     |            266836
modernize-use-trailing-return-type                          |            255116
readability-magic-numbers                                   |            138216
modernize-avoid-c-arrays                                    |            116741
-------------------------------------------------------------------------------
----=================----
```

### CODE REVIEW

**Review by bruntib — CHANGES_REQUESTED** [review]

File: analyzer/codechecker_analyzer/cmd/parse.py [hunk 1]
```diff
@@ -410,55 +425,81 @@ def get_output_file_path(default_file_name: str) -> Optional[str]:
...
-                plaintext.convert(
-                    file_report_map, processed_file_paths, print_steps)
-            elif export == 'html':
-                print(f"Parsing input file '{file_path}'.")
-                report_to_html.convert(
-                    file_path, reports, output_dir_path,
-                    html_builder)
-
-    if export is None:  # Plain text output
+            if 'summary' in args:
```
Comment (Szelethus) [inline analyzer/codechecker_analyzer/cmd/parse.py:428]: I'm not to hot about endlessly elongating functions like this, but I've seen arguments for it too, so, if you really insist, sure, but I'd prefer to have a dedicated function for this to lessen the scope of it. That would make unit testing easier as well.

On another note, #3843 does much of the same thing as this patch, can we not merge those implementations?

**Review by Szelethus — CHANGES_REQUESTED** [review]
We need tests before landing this. Maybe an `analyze_and_parse` test file like this?:
https://github.com/Ericsson/codechecker/blob/master/analyzer/tests/functional/analyze_and_parse/test_files/diagnostic_message_hash_clang_tidy.output

**Review by dkrupp — CHANGES_REQUESTED** [review]
Please add test cases. Otherwise looks ok.

**Review by dkrupp — CHANGES_REQUESTED** [review]
The output of the summary and the normal CodeChecker parse command differs.
I guess they should print the same.

I suspect the root cause is that we don't do bug deduplacation at the calculation of this summary.
This is related:
https://github.com/Ericsson/codechecker/issues/4042

This summary:

Checker name                                   | Number of reports
------------------------------------------------------------------
cppcoreguidelines-special-member-functions     | 23068            
misc-misplaced-const                           | 345              
bugprone-sizeof-expression                     | 335              
clang-diagnostic-implicit-int-float-conversion | 85               
clang-diagnostic-non-virtual-dtor              | 68               
clang-diagnostic-switch-enum                   | 65               
clang-diagnostic-implicit-int-conversion       | 48               
google-global-names-in-headers                 | 44               
cert-err09-cpp                                 | 37               
clang-diagnostic-implicit-fallthrough          | 13               
clang-diagnostic-misleading-indentation        | 13               
bugprone-misplaced-widening-cast               | 6                
clang-diagnostic-implicit-float-conversion     | 6                
clang-diagnostic-implicitly-unsigned-literal   | 4                
clang-diagnostic-double-promotion              | 3                
clang-diagnostic-unused-parameter              | 2                
clang-diagnostic-float-conversion              | 1                
clang-diagnostic-unused-variable               | 1                
misc-unconventional-assign-operator            | 1                
clang-diagnostic-sign-compare                  | 1  

Parse:
----==== Checker Statistics ====----
-----------------------------------------------------------------------------
Checker name                                   | Severity | Number of reports
-----------------------------------------------------------------------------
cppcoreguidelines-special-member-functions     | LOW      |               533
misc-misplaced-const                           | LOW      |                 7
clang-diagnostic-implicit-int-conversion       | MEDIUM   |                31
clang-diagnostic-switch-enum                   | MEDIUM   |                65
bugprone-sizeof-expression                     | HIGH     |                80
cert-err09-cpp                                 | HIGH     |                37
clang-diagnostic-implicit-fallthrough          | MEDIUM   |                13
clang-diagnostic-misleading-indentation        | MEDIUM   |                11
clang-diagnostic-float-conversion              | MEDIUM   |                 1
bugprone-misplaced-widening-cast               | HIGH     |                 3
clang-diagnostic-non-virtual-dtor              | MEDIUM   |                 3
clang-diagnostic-unused-parameter              | MEDIUM   |                 2
clang-diagnostic-implicit-int-float-conversion | MEDIUM   |                17
clang-diagnostic-unused-variable               | MEDIUM   |                 1
misc-unconventional-assign-operator            | MEDIUM   |                 1
clang-diagnostic-sign-compare                  | MEDIUM   |                 1
google-global-names-in-headers                 | STYLE    |                26
clang-diagnostic-implicit-float-conversion     | MEDIUM   |                 6
clang-diagnostic-double-promotion              | MEDIUM   |                 3
clang-diagnostic-implicitly-unsigned-literal   | MEDIUM   |                 4
-----------------------------------------------------------------------------

**Comment by cservakt** [comment]
This PR is not the appropriate implementation. After redesigning the report directory, there will be a correct solution to sum statistics.

**Pull request closed without merging**